### ISSUE TITLE
Time varying external fields

### ISSUE THREAD

**Issue by linigoga** [SmileiPIC/Smilei#648]
Hi everyone,

I am trying to add some time a Gaussian varying external fields to an input deck. Does SMILEI accept such type of fields? can the fields be both spatial and time dependent? and to add the fields for different directions, does it always have to add a new code block?

Thanks for you help! 
Best, 
Lucas

**Comment by mccoys** [comment]
It is not possible to force fields in a self-consistent simulation. What you can do is to choose (divergence-free) initial fields (this is called external fields in smilei) or choose fields at the boundaries (currently available as lasers in smilei).

There is a block called prescribed fields but it is not for self-consistent simulations. Only for test purposes. So you should not use it in general.

**Comment by linigoga** [comment]
Hi, 

Thank you for your reply! Ok. what I wanted to do is in a 2D simulation, add a field with a temporal gaussian profile similar to a laser like a laser propagating transverse to the simulation plane. I guess this not be possible?

Best,
Lucas

**Comment by mccoys** [comment]
I don't think it is possible with a plasma in there. If there is no plasma, or only test particles, you could try prescribed fields, but there is no point in using a pic code for that

**Comment by linigoga** [comment]
Right, it makes sense. thank you for your help!
